tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -514,15 +514,20 @@ private[optimizer] abstract class OptimizerCore(
       case RecordValue(tpe, elems) =>
         RecordValue(tpe, elems map transformExpr)
 
-      case IsInstanceOf(expr, ClassType(ObjectClass)) =>
+      case IsInstanceOf(expr, ClassRef(ObjectClass)) =>
         transformExpr(BinaryOp(BinaryOp.!==, expr, Null()))
 
-      case IsInstanceOf(expr, tpe) =>
+      case IsInstanceOf(expr, typeRef) =>
         trampoline {
           pretransformExpr(expr) { texpr =>
             val result = {
-              // TODO This cast is suspicious
-              if (isSubtype(texpr.tpe.base, tpe.asInstanceOf[Type])) {
+              // TODO The hard conversion of ClassRef into ClassType is suspicious
```

Comment:
I don't think this is suspicious looking at it now. If you look at how its used, it's simply a workaround to not implement `isSubRefType`. Completely acceptable if we document it (NB that `typeRefTpe` cannot be a JS type by construction, does the IR checker verify this?).